kedro/runner/runner.py

```diff
@@ -6,17 +6,31 @@
 
 import inspect
 import logging
+import os
+import sys
 import warnings
 from abc import ABC, abstractmethod
-from collections import deque
+from collections import Counter, deque
+from concurrent.futures import (
+    FIRST_COMPLETED,
+    ProcessPoolExecutor,
+    ThreadPoolExecutor,
+    wait,
```

Comment:
Would this import `multiprocessing` as a dependencies? I recalled in the past we have issues with `ShelveStore` because even importing the library cause issues on restricted environment like AWS Lambda.